### ISSUE TITLE
Refactor Action to a set of entities (one per action type) implementing a protocol

### ISSUE THREAD

**Issue by ctarda** [wordpress-mobile/WordPress-iOS#9638]
All actions used in Notifications and Activity Log should be abstracted behind a protocol.
- ApproveComment
- Follow
- LikeComment
- ReplyToComment
- SpamComment (mark as spam)
- TrashComment

Actions can be enabled or disabled, that might be a couple of methods in the interface

Actions also can be On or Off. Another couple of methods / properties

Each action might expose an identifier (actions are retained by a NotificationBlock in a dictionary)

Actions are leaking all the way up to the NotificationsViewController at the moment, who decides what method to call on the NotificationActionsService, according to the Action. We should encapsulate either that directly on the different actions. That should also make the current NotificationDeletionRequest unecessary

In other words, each implementation of the Action protocol would be responsible for executing a different method in the NotificationActionsService

This way,  Actions can also be reused in NotificationDetailsViewController. 

https://docs.google.com/document/d/1KS9hyrXZAoQYNQ-Ug2nSR0JmOMsj7wkHJwlkbmsW8Dk/edit?usp=sharing

**Comment by loremattei** [comment]
Closing it as it's solved by #9657.
